packages/core/src/internal/fetchSource.ts

```diff
@@ -175,9 +175,15 @@ async function* fetchOperation(
       throw new Error(await response.text());
     }
 
+    let pending: ExecutionResult['pending'];
     for await (const payload of results) {
+      if (payload.pending && !result) {
+        pending = payload.pending;
+      } else if (payload.pending) {
+        pending = [...pending!, ...payload.pending];
+      }
```

Comment:
Initially pending will be present on the ExecutionResult, however when we have nested incremental results then the IncrementalResult will contain new pending information which we have to add to our array here so we can keep building on our result